qstack/orcaio.py

```diff
@@ -38,8 +42,115 @@ def read_density(mol, basename, directory='./', version=500, openshell=False, re
+            warnings.warn(msg)
+
     if reorder_dest is not None:
         dm = np.array([reorder_ao(mol, i, src='orca', dest=reorder_dest) for i in dm])
 
     dm = np.squeeze(dm)
     return dm
+
+
+def _parse_gbw(fname):
+    """ Many thanks to
+    https://pysisyphus.readthedocs.io/en/latest/_modules/pysisyphus/calculators/ORCA.html
+    """
+
+    def read_array(f, n, dtype):
+        return np.frombuffer(f.read(dtype().itemsize * n), dtype=dtype)
+
+    def read_mos():
+        f.seek(24)
+        offset = struct.unpack("<q", f.read(np.int64().itemsize))[0]    # int64: pointer to orbitals
+        f.seek(offset)
+        sets   = struct.unpack("<i", f.read(np.int32().itemsize))[0]    # int32: number of MO sets
+        nao    = struct.unpack("<i", f.read(np.int32().itemsize))[0]    # int32: number of orbitals
+        assert sets in (1,2)
+
+        coefficients_ab = []
+        energies_ab = []
+        occupations_ab = []
+
+        for i in range(sets):
+            coefficients = read_array(f, nao*nao, np.float64).reshape(-1, nao)
+            occupations  = read_array(f, nao,     np.float64)
+            energies     = read_array(f, nao,     np.float64)
+            irreps       = read_array(f, nao,     np.int32)
+            cores        = read_array(f, nao,     np.int32)
+            coefficients_ab.append(coefficients)
+            energies_ab.append(energies)
+            occupations_ab.append(occupations)
+
+        coefficients_ab = np.array(coefficients_ab)
+        energies_ab     = np.array(energies_ab)
+        occupations_ab  = np.array(occupations_ab)
+        return coefficients_ab, energies_ab, occupations_ab
+
+    def read_basis(MAX_PRIMITIVES=37):
+        f.seek(16)
+        offset = struct.unpack("<q", f.read(np.int64().itemsize))[0]
+        f.seek(offset)
+        _, nat = struct.unpack("<2i", f.read(2 * np.int32().itemsize))
+        ls = {}
+        for at in range(nat):
+            ls[at] = []
+            _, nao_at = struct.unpack("<2i", f.read(2 * np.int32().itemsize))
+            for iao in range(nao_at):
+                l, _, ngto, _ = struct.unpack("<4i", f.read(4 * np.int32().itemsize))
+                a = read_array(f, MAX_PRIMITIVES, np.float64)   # exponents
+                c = read_array(f, MAX_PRIMITIVES, np.float64)   # coefficients
+                ls[at].append(l)
+        return ls
+
+    with open(fname, "rb") as f:
+        coefficients_ab, energies_ab, occupations_ab = read_mos()
+        ls = read_basis()
+        return coefficients_ab, energies_ab, occupations_ab, ls
+
+
+def reorder_coeff_inplace(c_full, mol, reorder_dest='pyscf', ls_from_orca=None):
+    def _reorder_coeff(c):
+        # In ORCA, at least def2-SVP and def2-TZVP for 3d metals
+        # are not sorted wrt angular momenta
+        # TODO add the fix to read_density()   #if gto.basis._format_basis_name(mol.basis)=='def2tzvp':
```

Comment:
I think you did that (and can remove the `TODO`)!
_unless you wanted to do something more than throwing a `warn()`_